application-ckeditor-plugins/src/main/resources/xwiki-image/imageSelector.js

```diff
@@ -167,4 +122,10 @@ define('imageSelector', ['jquery', 'modal', 'resource', 'l10n!imageSelector'],
         });
       }
     });
+    return {
+      createModal: createModal,
+      saveSelectedImageReference: saveSelectedImageReference,
+      removeSelectedImageReference: removeSelectedImageReference,
+      getEditor: getEditor
```

Comment:
I don't like this. I think the image selector should not depend on the editor. We should be able (in the future) to use the image selector from any editor (Wiki, Inline Form) or even from view mode.

Reply:
Indeed, in this case we need a way to abstract over the upload operations, I'll see what I can do.
(I only exposed the editor to implement the logic of `Upload.xml`).

Reply:
`createLoader` provided instead. The method take in parameter the upload field and an option object with callbacks.